package/MDAnalysis/analysis/rms.py

```diff
@@ -398,125 +416,100 @@ def __init__(self, traj, reference=None, select='all',
+                    select_atoms(*s['reference']).
+                    positions.astype(np.float64)) for s in
                     self.groupselections]
         finally:
             # Move back to the original frame
             self.reference.trajectory[current_frame]
-        ref_coordinates_64 = ref_coordinates.astype(np.float64)
 
-        # allocate the array for selection atom coords
-        traj_coordinates = traj_atoms.positions.copy()
+        self._ref_coordinates_64 = self._ref_coordinates.astype(np.float64)
 
-        if self.groupselections_atoms:
+        if self._groupselections_atoms:
             # Only carry out a rotation if we want to calculate secondary
             # RMSDs.
             # R: rotation matrix that aligns r-r_com, x~-x~com
             #    (x~: selected coordinates, x: all coordinates)
             # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
-            rot = np.zeros(9, dtype=np.float64)  # allocate space
-            R = np.matrix(rot.reshape(3, 3))
+            self._rot = np.zeros(9, dtype=np.float64)  # allocate space
+            self._R = np.matrix(self._rot.reshape(3, 3))
+        else:
+            self._rot = None
+
+        self.rmsd = np.zeros((self.n_frames,
+                              3 + len(self._groupselections_atoms)))
+
+        self._pm.format = ("RMSD %(rmsd)5.2f A at frame "
+                           "%(step)5d/%(numsteps)d  [%(percentage)5.1f%%]\r")
+
+    def _single_frame(self):
+        mobile_com = self.mobile_atoms.center_of_mass().astype(np.float64)
+        mobile_coordinates = self.mobile_atoms.positions - mobile_com
+        self.rmsd[self._frame_index, :2] = self._ts.frame, self._trajectory.time
+
+        if self._groupselections_atoms:
+            # 1) superposition structures. No longer need to transpose coordinates
+            # such that the coordinate array is 3xN instead of Nx3. Also
+            # qcp requires that the dtype be float64 (I think we swapped
+            # the position of ref and traj in CalcRMSDRotationalMatrix so
+            # that R acts **to the left** and can be broadcasted; we're
+            # saving one transpose. [orbeckst])
+            self.rmsd[self._frame_index, 2] = qcp.CalcRMSDRotationalMatrix(
+                self._ref_coordinates_64,
+                mobile_coordinates.astype(np.float64), self._n_atoms,
+                self._rot, self._weights)
+
+            #logger.info('mobile atom positions {}'.format(self.mobile_atoms.positions[0]))
+            #logger.info('self.ref_coordinates_64 {}'.format(self._ref_coordinates_64[0]))
+            self._R[:,:] = self._rot.reshape(3, 3)
+            # Transform each atom in the trajectory (use inplace ops to
+            # avoid copying arrays) (Marginally (~3%) faster than
+            # "ts.positions[:] = (ts.positions - x_com) * R + ref_com".)
+            self.mobile_atoms.positions -= mobile_com
```

Comment:
**This is incorrect.** You need to transform *all* atoms, not only the mobile selection. If, for instance, your mobile selection is a protein A, and your group is a ligand bound to A, then if you only transform mobile, the ligand will remain at its old positions and you will compute a completely wrong RMSD.

The reason why the tests pass within 2-3 decimals is, I suppose, because the test trajectory is already almost aligned.

The original code correctly transformed all positions *in the timestep*, `ts.positions`. This should be 
```python
positions = self.universe.trajectory.ts.positions
positions -= mobile_com
positions[:] = positions * R
positions += self._ref_com
```

What this does is the standard superposition procedure (see the [RMS fitting tutorial](http://www.mdanalysis.org/mdanalysis/documentation_pages/analysis/align.html#rms-fitting-tutorial)):
  1. You shift the origin of the target coordinates to origin (because we already shifted the reference coordinates so that its center of mass is also at the origin). In this way you remove all translations.
  2. Apply the pure rotation (which was calculated earlier).
  3. Shift the mobile coordinates to the original position of the *reference* so that the translational superposition is satisfied.